### PR TITLE
IBC refactor - Zero Diff,  rename types and methods 

### PR DESCRIPTION
**Pull request by briansull**
Rename struct ICorJitInfo::ProfileBuffer        to ICorJitInfo::BlockCounts
Rename method ICorJitInfo::allocBBProfileBuffer to ICorJitInfo::allocMethodBlockCounts
Rename method ICorJitInfo::getBBProfileData     to ICorJitInfo:"getMethodBlockCounts
Rename args and use DWORD instead of ULONG for ICorJitInfo:allocMethodBlockCounts and ICorJitInfo:getMethodBlockCounts
Rename Compiler::fgProfileBuffer  to Compiler::fgBlockCounts
Use an #ifdef FEATURE_CORECLR to fix the missing CORINFO_FLG_DISABLE_TIER0_FOR_LOOPS flag on the desktop.
Make fgBlockCountsCount and fgNumProfileRuns DWORDs instead of ULONGs
Rename local var bbCurrentBlockProfileBuffer to currentBlockCounts
Rename local var bbProfileBufferStart to profileBlockCountsStart
Use DWORD when iterating over BlockCounts instead of ULONG
Rename ZapImage::hashBBProfileData to ZapImage::hashMethodBlockCounts
SuperPMI - Fixed all references to allocBBProfileBuffer => allocMethodBlockCounts
SuperPMI - fixed all reference to getBBProfileBuffer => getMethodBlockCounts

### CODE REVIEW

File: src/jit/compiler.h [hunk 1]
```diff
@@ -5323,10 +5323,10 @@ class Compiler
 
     void fgAdjustForAddressExposedOrWrittenThis();
 
-    bool                        fgProfileData_ILSizeMismatch;
-    ICorJitInfo::ProfileBuffer* fgProfileBuffer;
-    ULONG                       fgProfileBufferCount;
-    ULONG                       fgNumProfileRuns;
+    bool                      fgProfileData_ILSizeMismatch;
+    ICorJitInfo::BlockCounts* fgBlockCounts;
+    DWORD                     fgBlockCountsCount;
```
Comment (mikedn) [inline src/jit/compiler.h]: Why the change from ULONG to DWORD? DWORD is not commonly used in the JIT codebase for "count". In fact neither is, it's more common to use `unsigned`.

Comment (briansull) [reply src/jit/compiler.h]: I hate ULONG (because it is 32-bits in size), and DWORD is used by the original definition.

src/inc/corbbtprof.h
```
struct CORBBTPROF_BLOCK_DATA    // This struct is also defined by:  ICorJitInfo.BlockCounts
{	{
    DWORD                          ILOffset;	    DWORD                          ILOffset;
    DWORD                          ExecutionCount;	    DWORD                          ExecutionCount;
};	};

```

Comment (briansull) [reply src/jit/compiler.h]: I prefer UINT32 or unsigned

**Comment by briansull** [comment]
Updated PR: Prefer UINT32 over DWORD in the IBC BlockCount API's

**Comment by briansull** [comment]
@dotnet/jit-contrib  PTAL

**Comment by briansull** [comment]
@dotnet/jit-contrib PTAL

File: src/jit/compiler.cpp [hunk 2]
```diff
@@ -5948,7 +5948,12 @@ int Compiler::compCompileHelper(CORINFO_MODULE_HANDLE            classPtr,
         goto _Next;
     }
 
+#ifdef FEATURE_CORECLR
     if (fgHasBackwardJump && (info.compFlags & CORINFO_FLG_DISABLE_TIER0_FOR_LOOPS) != 0 && fgCanSwitchToTier1())
```
Comment (sandreenko) [inline src/jit/compiler.cpp]: The condition is a bit long, we can merge it with `!FEATURE_CORECLR` or use the same order for them:
`if (fgHasBackwardJump && fgCanSwitchToTier1() && ((info.compFlags & CORINFO_FLG_DISABLE_TIER0_FOR_LOOPS) != 0))`
